transport/peerlist.go

```diff
@@ -0,0 +1,15 @@
+package transport
+
+import "context"
+
+//go:generate mockgen -destination=transporttest/peerlist.go -package=transporttest go.uber.org/yarpc/transport PeerList
+
+// PeerList is a collection of Peers.  Outbounds request peers from the ChoosePeer to determine where to send requests
+type PeerList interface {
+	PeerSubscriber
+
+	Start() error // Notify the PeerList that it will start receiving requests
+	Stop() error  // Notify the PeerList that it will stop receiving requests
```

Comment:
nit: for method comments, put them above the function declaration rather than to the right